pilot/pkg/networking/grpcgen/lds.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcgen
+
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/util"
+)
+
+// BuildListeners handles a LDS request, returning listeners of ApiListener type.
+// The request may include a list of resource names, using the full_hostname[:port] format to select only
+// specific services.
+func (g *GrpcConfigGenerator) BuildListeners(node *model.Proxy, push *model.PushContext, names []string) model.Resources {
+	resp := model.Resources{}
+
+	filter := map[string]bool{}
+	for _, name := range names {
+		if strings.Contains(name, ":") {
+			n, _, err := net.SplitHostPort(name)
+			if err == nil {
+				name = n
+			}
+		}
+		filter[name] = true
```

Comment:
Why is port not part of the key? Seems like we return ports they didn't ask for?

Reply:
Yeah this can be cleaned up – lds is 99% copypaste from the original code into a new file